File: frontend/src/api/chat/dto/ChatMessage.ts
export type MessageType = 'CHAT' | 'JOIN' | 'LEAVE'

export interface ChatMessageRequest {
  rentalItemId: number
  receiverId: number
  content: string
}

export interface ChatMessageResponse {
  id: string
  chatRoomId: number
  senderId: number
  senderNickname: string
  content: string
  type: MessageType
  timestamp: string
}

File: backend/api/src/main/kotlin/io/github/daegwonkim/backend/websocket/ChatMessage.kt
package io.github.daegwonkim.backend.websocket

import java.time.Instant
import java.util.UUID

enum class MessageType {
    CHAT,
    JOIN,
    LEAVE
}

data class ChatMessageRequest(
    val roomId: Long,
    val content: String
)

data class ChatMessageResponse(
    val id: UUID = UUID.randomUUID(),
    val roomId: Long,
    val senderId: Long,
    val senderNickname: String,
    val content: String,
    val type: MessageType,
    val timestamp: Instant = Instant.now()
)
